src/core/utils/wrap_stdbool.h

```diff
@@ -44,7 +44,7 @@
 /* c++ has a built in bool */
 #else
 
-#if defined(_MSC_VER)
+#if HAVE__BOOL
```

Comment:
First, there aren't two versions of VS here. The difference is user-mode (using SDK) and kernel-mode (using WDK) on Windows. I don't know why exactly, but certain 'standard' headers are not available in kernel mode; possibly just for legacy reasons.

Also, we don't need two different config headers. The existing [openthread-windows-config.h](https://github.com/openthread/openthread/blob/master/include/openthread-windows-config.h#L103) already had `#ifdef _KERNEL_MODE` blocks that control differences between the two. We could put the `HAVE__BOOL` define in there, but that would require the config file to be included first. I'm not sure how big a change that would be.